spec/latest/core/core.jsonld

```diff
@@ -253,7 +254,8 @@
       "label": "expects",
       "comment": "The information expected by the Web API.",
       "domain": "hydra:Operation",
-      "range": "hydra:Class",
+      "range": "hydra:Resource",
+      "rangeIncludes": ["hydra:Resource", "hydra:Class"],

```

Comment:
I think it's redundant to repeat `hydra:Resource` here. In fact, a `Resource` itself is not useful as an object of `expects`/`returns`. It has to be something more specific, like `Class`

```suggestion
      "rangeIncludes": ["hydra:Class"],
```

Reply:
I don't agree. I think it's useful to have both predicates to contain `hydra:Resource`. Previously used `rdfs:range` is more for compatibility, but I'd like to use `schema:rangeIncludes` in future implementations - it has weaker semantics.

I'd leave it as I suggested, at least untill some more specific alternatives are available.

Reply:
Isn't `hydra:Resource` what enables non-RDF payloads?

Reply:
@asbjornu, `schema:rangeIncludes` doesn't have that strong semantics as `rdfs:range`. Putting only `hydra:Class` inside `schema:rangeIncludes` doesn't infer no other types can be used - it's more of a hint (well the name has it - range includes, but not restricts). Still, I'd prefer to have both so clients doesn't have to concatenate both `rdfs:range` and `schema:rangeIncludes` to discover what's actually allowed.

Reply:
I see. Thanks for the explanation, @alien-mcl!

Reply:
But `Resource` is range includes is actually not truthful. Most classes of Hydra are subclasses of `hydra:Resource` but they will not make sense as `expects`/`returns`. This hint is misleading.